admin/class-yoast-network-admin.php

```diff
@@ -0,0 +1,248 @@
+<?php
+/**
+ * WPSEO plugin file.
+ *
+ * @package WPSEO\Internals
+ */
+
+/**
+ * Multisite utility class for network admin functionality.
+ */
+class Yoast_Network_Admin {
+
+	/**
+	 * Action identifier for updating plugin network options.
+	 */
+	const UPDATE_OPTIONS_ACTION = 'yoast_handle_network_options';
+
+	/**
+	 * Action identifier for restoring a site.
+	 */
+	const RESTORE_SITE_ACTION = 'yoast_restore_site';
+
+	/**
+	 * @var bool Internal flag for whether the necessary hooks have been added.
+	 */
+	private $hooks_registered = false;
```

Comment:
I would remove this. We are setting the hooks once and if somebody else is using this class, it isn't our responsibility to check if hooks are set or not.